cli/http_util.rs

```diff
@@ -75,77 +72,45 @@ pub enum FetchOnceResult {
 pub fn fetch_string_once(
   url: &Url,
 ) -> impl Future<Output = Result<FetchOnceResult, ErrBox>> {
-  type FetchAttempt = (Option<String>, Option<String>, Option<FetchOnceResult>);
-
   let url = url.clone();
   let client = get_client();
 
-  futures::compat::Compat01As03::new(client.get(url.clone()).send())
-    .map_err(ErrBox::from)
-    .and_then(
-      move |mut response| -> Pin<
-        Box<dyn Future<Output = Result<FetchAttempt, ErrBox>> + Send>,
-      > {
-        if response.status().is_redirection() {
-          let location_string = response
-            .headers()
-            .get(LOCATION)
-            .expect("url redirection should provide 'location' header")
-            .to_str()
-            .unwrap();
-
-          debug!("Redirecting to {:?}...", &location_string);
-          let new_url = resolve_url_from_location(&url, location_string);
-          // Boxed trait object turns out to be the savior for 2+ types yielding same results.
-          return futures::future::try_join3(
-            future::ok(None),
-            future::ok(None),
-            future::ok(Some(FetchOnceResult::Redirect(new_url))),
-          )
-          .boxed();
-        }
-
-        if response.status().is_client_error()
-          || response.status().is_server_error()
-        {
-          return future::err(
-            DenoError::new(
-              deno_error::ErrorKind::Other,
-              format!("Import '{}' failed: {}", &url, response.status()),
-            )
-            .into(),
-          )
-          .boxed();
-        }
-
-        let content_type = response
-          .headers()
-          .get(CONTENT_TYPE)
-          .map(|content_type| content_type.to_str().unwrap().to_owned());
-
-        let body = futures::compat::Compat01As03::new(response.text())
-          .map_ok(Some)
-          .map_err(ErrBox::from);
-
-        futures::future::try_join3(
-          body,
-          future::ok(content_type),
-          future::ok(None),
-        )
-        .boxed()
-      },
-    )
-    .and_then(move |(maybe_code, maybe_content_type, maybe_redirect)| {
-      if let Some(redirect) = maybe_redirect {
-        future::ok(redirect)
-      } else {
-        // maybe_code should always contain code here!
-        future::ok(FetchOnceResult::Code(
-          maybe_code.unwrap(),
-          maybe_content_type,
-        ))
-      }
-    })
+  let fut = async move {
+    let response = client.get(url.clone()).send().await?;
```

Comment:
```suggestion
    let response = client.get(url).send().await?;
```

Reply:
Doesn't compile without this clone:
```
error[E0382]: borrow of moved value: `url`
  --> cli/http_util.rs:90:47
   |
79 |     let response = client.get(url).send().await?;
   |                               --- value moved here
...
90 |       let new_url = resolve_url_from_location(&url, location_string);
   |                                               ^^^^ value borrowed here after move
   |
   = note: move occurs because `url` has type `url::Url`, which does not implement the `Copy` trait

```